torch/csrc/jit/codegen/cuda/codegen.cpp

```diff
@@ -294,9 +305,7 @@ class CudaKernelGenerator : private kir::IrVisitor {
     }
   }
 
-  void visit(const kir::TensorIndex* node) final {
-    code_ << varName(node->view()) << "[";
-
+  void gen1DStridedTensorIndex(const kir::TensorIndex* node) {
```

Comment:
Original tensor index is renamed as `1D` index since the strides were already applied to the indices so they just needed to sum up.

Reply:
Yes we can build that out in a separate PR. It'd be quite a bit of detail there to support this through index compute. For gemm we will need to do something like:
```cpp
iterdomain sx,sy,x,y;
sx,sy = swizzle(x,y)
sxo,sxi = split(sx)
syo,syi = split(sy)
...
```

Reply:
But I do agree it will be much cleaner and more powerful if we do these through index compute.